app/js/directives.js

```diff
@@ -432,28 +432,38 @@ angular.module('myApp.directives', ['myApp.filters'])
       templateUrl: templateUrl('message_attach_contact')
     }
   })
-  .directive('myMessageWebpage', function (AppWebPagesManager, AppPhotosManager) {
+  .directive('myMessageWebpage', function (AppWebPagesManager, AppPhotosManager, Storage) {
     return {
       scope: {
         'media': '=myMessageWebpage',
         'messageId': '=messageId'
       },
       templateUrl: templateUrl('message_attach_webpage'),
       link: function ($scope) {
-        $scope.openPhoto = AppPhotosManager.openPhoto
-        $scope.openEmbed = function ($event) {
-          if ($scope.media.webpage &&
-            $scope.media.webpage.embed_url) {
-            AppWebPagesManager.openEmbed($scope.media.webpage.id, $scope.messageId)
-            return cancelEvent($event)
-          }
-        }
+        Storage.get('disable_webpage_preview').then(function (previewDisabled) {
+          $scope.previewDisabled = !!previewDisabled;
```

Comment:
yes you're right but when the app is first run, the storage value is not going to exist and that should give some weird value i.e: `undefined` or `null` depending on how the `Storage` service is built. doing `!!` should cover for that case too and give a solid `Boolean` type. Unless the storage service already handles it? I didn't look under the hood of that service.